File: src/main/java/de/danielmaile/aether/worldgen/BedrockPopulator.java
package de.danielmaile.aether.worldgen;

import org.bukkit.Material;
import org.bukkit.generator.BlockPopulator;
import org.bukkit.generator.LimitedRegion;
import org.bukkit.generator.WorldInfo;
import org.jetbrains.annotations.NotNull;

import java.util.Random;

public class BedrockPopulator extends BlockPopulator
{
    @Override
    public void populate(@NotNull WorldInfo worldInfo, @NotNull Random random, int chunkX, int chunkZ, @NotNull LimitedRegion limitedRegion)
    {
        for (int x = chunkX * 16; x < (chunkX + 1) * 16; x++)
        {
            for (int z = chunkZ * 16; z < (chunkZ + 1) * 16; z++)
            {
                limitedRegion.setType(x, -256, z, Material.BEDROCK);
            }
        }
    }
}

File: src/main/java/de/danielmaile/aether/world/BedrockPopulator.kt
package de.danielmaile.aether.world

import org.bukkit.Material
import org.bukkit.generator.BlockPopulator
import org.bukkit.generator.LimitedRegion
import org.bukkit.generator.WorldInfo
import java.util.*

class BedrockPopulator : BlockPopulator() {

    override fun populate(
        worldInfo: WorldInfo,
        random: Random,
        chunkX: Int,
        chunkZ: Int,
        limitedRegion: LimitedRegion
    ) {
        for (x in chunkX * 16 until (chunkX + 1) * 16) {
            for (z in chunkZ * 16 until (chunkZ + 1) * 16) {
                limitedRegion.setType(x, -256, z, Material.BEDROCK)
            }
        }
    }
}